be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -586,7 +594,61 @@ Status HdfsOrcScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
 Status HdfsOrcScanner::do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     _should_skip_file = false;
     _use_orc_sargs = true;
+    build_row_delete_set(scanner_params);
     // todo: build predicate hook and ranges hook.
     return Status::OK();
 }
+
+Status HdfsOrcScanner::build_row_delete_set(const HdfsScannerParams& scanner_params) {
+    for (const auto& tdelete_file : scanner_params.deletes) {
+        scan_pos_delete_file(*tdelete_file);
+    }
+
+    return Status::OK();
+}
+
+Status HdfsOrcScanner::scan_pos_delete_file(const TIcebergDeleteFile& delete_file) {
+    std::unique_ptr<RandomAccessFile> file;
+    ASSIGN_OR_RETURN(file, _scanner_params.fs->new_random_access_file(delete_file.full_path));
+    auto input_stream = std::make_unique<ORCHdfsFileStream>(file.get(), delete_file.length);
+    std::unique_ptr<orc::Reader> reader;
+    try {
+        orc::ReaderOptions options;
+        reader = orc::createReader(std::move(input_stream), options);
+    } catch (std::exception& e) {
+        auto s = strings::Substitute("createReader throws. reason = $0", e.what());
+        LOG(WARNING) << s;
+        return Status::InternalError(s);
+    }
+    auto row_reader = reader->createRowReader(orc::RowReaderOptions{});
+    auto batch = row_reader->createRowBatch(1024);
+
+    try {
+        while (row_reader->next(*batch)) {
+            auto* struct_batch = down_cast<orc::StructVectorBatch*>(batch.get());
+
+            // position-delete file_path
+            auto* path_batch = down_cast<orc::StringVectorBatch*>(struct_batch->fields[0]);
+            // position-delete pos
+            auto* pos_batch = down_cast<orc::LongVectorBatch*>(struct_batch->fields[1]);
+
+            for (size_t row = 0; row < path_batch->numElements; row++) {
+                std::string s(path_batch->data[row], path_batch->length[row]);
```

Comment:
Can we get the relevant deleted files on fe?

Reply:
yes, but one delete file may be related to multiple data file
so deleted row should be located by (file_path, pos)